File: Frontend/src/types/user/coupons/dtos/SearchActiveCouponsQuery.ts
import type { PagedAndSortedRequest } from "../../../common/dtos/PagedAndSortedRequest";
import { DISCOUNT_TYPE } from "../enums/DiscountType";
import { COUPON_SORT_COLUMN } from "../enums/CouponSortColumn";

export interface SearchActiveCouponsQuery extends PagedAndSortedRequest {
  discountType?: (typeof DISCOUNT_TYPE)[keyof typeof DISCOUNT_TYPE];
  sortColumn?: (typeof COUPON_SORT_COLUMN)[keyof typeof COUPON_SORT_COLUMN];
}

File: Backend/Application/Features/Coupons/Queries/SearchActiveCoupons/SearchActiveCouponsQuery.cs
using Application.Common.Models;
using Application.Features.Coupons.DTOs;
using Application.Features.Coupons.Enums;
using Domain.Enums;
using Domain.Enums.Coupons;
using MediatR;

namespace Application.Features.Coupons.Queries.SearchActiveCoupons;

public class SearchActiveCouponsQuery : IRequest<PagedAndSortedResult<CouponDto>>
{
    public string? SearchText { get; set; }
    public int? PageIndex { get; set; }
    public int? PageSize { get; set; }
    public DiscountType? DiscountType { get; set; }
    public SortDirection? SortDirection { get; set; }
    public CouponSortColumn? SortColumn { get; set; }
}